packages/vuetify/src/components/VIcon/VIcon.ts

```diff
@@ -73,17 +73,19 @@ const VIcon = mixins(
     },
     // Component data for both font and svg icon.
     getDefaultData (): VNodeData {
+      const hasClickListener = Boolean(this.$listeners.click || this.$listeners['!click'])
       const data: VNodeData = {
         staticClass: 'v-icon',
         class: {
           'v-icon--disabled': this.disabled,
           'v-icon--left': this.left,
-          'v-icon--link': this.$listeners.click || this.$listeners['!click'],
+          'v-icon--link': hasClickListener,
           'v-icon--right': this.right,
           'v-icon--dense': this.dense
         },
         attrs: {
-          'aria-hidden': true,
+          'aria-hidden': !hasClickListener,
+          role: hasClickListener ? 'button' : null,
```

Comment:
maybe `aria-role`?

Reply:
I think it should be `role`

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

Also, I noticed, that in such case there should be `tabindex` - to allow icon to be selectable